### ISSUE TITLE
Protocol matching doesn't reject explicit ClassVar in class when protocol is not marked as ClassVar

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6869]
```python
from typing import ClassVar, Protocol, Sequence


class Proto(Protocol):
    val1: Sequence[int]

class Concrete:
    val1: ClassVar[Sequence[int]] = [0]

x: Proto = Concrete()  # Should generate an error
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.345, which I just published. It will also be included in a future release of pylance.
